joschahenningsen/TUM-Live
# 1171 seeking 15 second is too long

### Motivation and Context


Issue: https://github.com/TUM-Dev/gocast/issues/1171
The seeking time of a video was set to 15 second and it can not be changed. Sometimes to hear what just said 15 seconds can be too much. 


### Description

There used to be no user setting option related with seeking. Now the seekingTime is also added to the user settings. A button to choose the seeking time is added to the settings page of user. It is still set to 15 seconds as default, but now the user has also the options of 5 or 10 seconds a seeking Time. When the popular video platform on the web are examined, the general seeking options were 5, 10 and 15 seconds. Therefore there is no more option is added, but can always be added just by expanding the button because backend is already set. The option of whether allowing user to write his/her own seeking time can be discussed, but thought that these three options are enough to solve the issue and also giving concrete options in a rather small matter seemed more appropriate. 

The created button changes the new user setting SeekingTime. For seeking then the new value is used. Since user settings can store a value as a String, the conversion from String to a number/int was done in the backend, not in the frontend of the code. It is chosen so, because the same method is also used for playbackspeeds and making this conversion on the backend seemed more robust.

For getSeekTime and updateSeekTime methods, the already implemented get and update methods of the other user settings are considered and these methods were created accordingly. 

### Steps for Testing

Prerequisites:
Having a tumlive account

1. Log in
2. Navigate to settings
3. Change the seeking time
4. Navigate a live stream or to an already published video
5. Seek the video and see if it seeks 15 seconds or accordingly to what was chosen in settings
6. Log out
7. Log in again and control if the seeking time was still set to the same value you have chosen before

### Screenshots

![Settings_wit_Seeking_Tİme](https://github.com/TUM-Dev/gocast/assets/70471039/d1792d26-24ba-4d0a-8b07-e709ccd6e0e6)
